core/types/receipt.go

```diff
@@ -54,25 +56,58 @@ func NewReceipt(root []byte, cumulativeGasUsed *big.Int) *Receipt {
 	return &Receipt{PostState: common.CopyBytes(root), CumulativeGasUsed: new(big.Int).Set(cumulativeGasUsed), Status: TxStatusUnknown}
 }
 
+type receiptRLP struct {
+	PostStateOrStatus []byte
+	CumulativeGasUsed *big.Int
+	Bloom             Bloom
+	Logs              vm.Logs
+}
+
 // EncodeRLP implements rlp.Encoder, and flattens the consensus fields of a receipt
 // into an RLP stream.
 func (r *Receipt) EncodeRLP(w io.Writer) error {
-	return rlp.Encode(w, []interface{}{r.PostState, r.CumulativeGasUsed, r.Bloom, r.Logs})
+	var stateOrStatus []byte
+	if len(r.PostState) == 0 { // if there is no state, EIP-658 is in place
```

Comment:
Seems kind of weird that `bytes = nil` will have len `0`, but true (https://play.golang.org/p/94KAfTm_dEU).

Reply:
https://golang.org/pkg/builtin/#len:
> if v is nil, len(v) is zero.

Reply:
Yep. Just sayin'. Although IDK what the heck else the `len` should be ;)